components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split_ab2.F

```diff
@@ -0,0 +1,3257 @@
+
+            enddo
+            normalBarotropicVelocity(iEdge) = &
+                  normalThicknessFluxSum/layerThicknessSum
+
+            ! normalBaroclinicVelocity = normalVelocity -
+            !     normalBarotropicVelocity
+            do k = minLevelEdgeBot(iEdge), kmax
+               normalBaroclinicVelocity(k,iEdge) = &
+                         normalVelocity(k,iEdge) - &
+                 normalBarotropicVelocity(iEdge)
+            enddo
+
+            ! normalBaroclinicVelocity=0,
+            ! normalVelocity=0 on land cells
+            do k = kmax+1, nVertLevels
+               normalBaroclinicVelocity(k,iEdge) = 0.0_RKIND
+               normalVelocity(k,iEdge) = 0.0_RKIND
+            enddo
+         enddo ! edge loop
+
+         if (config_filter_btr_mode) then
+            ! filter normalBarotropicVelocity out of initial condition
+
+            normalVelocity(:,:) = normalBaroclinicVelocity(:,:)
+            normalBarotropicVelocity(:) = 0.0_RKIND
+
+         endif
+
+
+      end if ! not restart
+
+      ! Create reusable halo exchange for subcycling
+      call mpas_dmpar_exch_group_create(domain, subcycleGroupName)
+      call mpas_dmpar_exch_group_add_field(domain, subcycleGroupName, &
+                                           'sshSubcycle')
+      call mpas_dmpar_exch_group_add_field(domain, subcycleGroupName, &
+                                  'normalBarotropicVelocitySubcycle')
+      call mpas_dmpar_exch_group_build_reusable_buffers(domain, &
+                                                     subcycleGroupName)
+
+      !-----------------------------------------------------------------
+      if (config_use_self_attraction_loading) then
+         ! set ssh_sal_on to 1
+         ssh_sal_on = 1
+      else
+         ssh_sal_on = 0
+      endif
+
+      self_attraction_and_loading_beta = config_self_attraction_and_loading_beta
+
+      !-----------------------------------------------------------------
+      if ( config_do_restart ) then
+         block => domain % blocklist
+         call mpas_pool_get_subpool(block%structs, 'state', statePool)
+         call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+         call mpas_pool_get_subpool(block%structs, 'mesh', meshPool)
+
+         call mpas_pool_get_dimension(block % dimensions, 'nVertLevels', nVertLevels)
+         call mpas_pool_get_dimension(block % dimensions, 'nEdges', nEdges)
+
+         call mpas_pool_get_array(statePool, 'normalVelocity', &
+                                              normalVelocity, 1)
+
+         call mpas_pool_get_array(meshPool, 'minLevelEdgeBot', minLevelEdgeBot)
+         call mpas_pool_get_array(meshPool, 'maxLevelEdgeTop', maxLevelEdgeTop)
+
+         do iEdge = 1, nEdges
+            kmax  = maxLevelEdgeTop(iEdge)
+            ! normalVelocity=0 on land cells
+            do k = kmax+1, nVertLevels
+               normalVelocity(k,iEdge) = 0.0_RKIND
+            enddo
+         enddo ! edge loop
```

Comment:
I think you need to do this over the inactive cells at the top of the domain, too.  You are probably not testing with configurations that use this feature, so it probably doesn't show up as a problem.
```suggestion
         do iEdge = 1, nEdges
            ! normalVelocity=0 on land cells
            do k = 1, minLevelEdgeBog(iEdge)-1
               normalVelocity(k,iEdge) = 0.0_RKIND
            enddo
            kmax  = maxLevelEdgeTop(iEdge)
            do k = kmax+1, nVertLevels
               normalVelocity(k,iEdge) = 0.0_RKIND
            enddo
         enddo ! edge loop
```

Reply:
@xylar, Thanks a lot! I'm running the nightly suite using with my fix, but I'm going to add this and test again.

And you are right. My test on Compy was using the default time integration. I just confirmed it. I think I made a mistake when I recloned and recompiled... Thanks again.

Reply:
@hyungyukang, your changes look perfect.  Thanks!